### PR TITLE
add request type

### PR DESCRIPTION
**Pull request by dbu**
replace #452

### CODE REVIEW

File: src/SymfonyCache/EventDispatchingHttpCache.php [hunk 1]
```diff
@@ -129,16 +130,17 @@ protected function invalidate(Request $request, $catch = false)
...
+     * @param string        $name        Name of the event to trigger. One of the constants in FOS\HttpCache\SymfonyCache\Events
      * @param Request       $request
-     * @param Response|null $response If already available
+     * @param Response|null $response    If already available
+     * @param int           $requestType The request type (default KernelInterface::MASTER_REQUEST)
      *
      * @return Response The response to return, which might be provided/altered by a listener
      */
-    protected function dispatch($name, Request $request, Response $response = null)
+    protected function dispatch($name, Request $request, Response $response = null, $requestType = KernelInterface::MASTER_REQUEST)
```
Comment (Toflar) [inline src/SymfonyCache/EventDispatchingHttpCache.php]: Should use `HttpKernelInterface` here too

File: tests/Unit/SymfonyCache/CacheEventTest.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    public function testEventGetters()
+    {
+        $request = Request::create('/');
+
+        $event = new CacheEvent($this->kernel, $request);
+
+        $this->assertSame($this->kernel, $event->getKernel());
+        $this->assertSame($request, $event->getRequest());
+        $this->assertNull($event->getResponse());
+        $this->assertSame(KernelInterface::MASTER_REQUEST, $event->getRequestType());
```
Comment (Toflar) [inline tests/Unit/SymfonyCache/CacheEventTest.php]: Same here, use `HttpKernelInterface`.

File: tests/Unit/SymfonyCache/CacheEventTest.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+        $response = new Response();
+
+        $event = new CacheEvent($this->kernel, $request, $response, HttpKernelInterface::SUB_REQUEST);
+
+        $this->assertSame($this->kernel, $event->getKernel());
+        $this->assertSame($request, $event->getRequest());
+        $this->assertSame($response, $event->getResponse());
+        $this->assertSame(KernelInterface::SUB_REQUEST, $event->getRequestType());
```
Comment (Toflar) [inline tests/Unit/SymfonyCache/CacheEventTest.php]: And here :)

File: src/SymfonyCache/EventDispatchingHttpCache.php (hunk 1 above)
Comment (dbu) [reply src/SymfonyCache/EventDispatchingHttpCache.php]: right, fixed

**Comment by dbu** [comment]
argh. github squashing stole the commit from you, sorry!

**Comment by Toflar** [comment]
As long as I do get the code changes I need, I don't really care who did what 😄 It was a minor change anyway.

**Comment by dbu** [comment]
i tagged 2.7.0

**Pull request merged**